src/requests/GroupsApi.ts

```diff
@@ -0,0 +1,1246 @@
+
+
+        if (userId.length <= 0) return Promise.reject(new Error('userId is too short!'));
+
+        const body: VRCAPI.Generics.dataKeysGroupBanMember = {
+            userId: userId,
+        }
+        const paramRequest: VRCAPI.Generics.executeRequestType = {
+            currentRequest: ApiPaths.groups.banGroupMember,
+            pathFormated: ApiPaths.groups.banGroupMember.path.replace('{groupId}', groupId),
+            body: body
+        };
+
+        this.checkValidData(paramRequest);
+        const queryResult = await this.executeRequest<VRCAPI.Groups.Models.GroupMember>(paramRequest);
+
+        return queryResult;
+
+    }
+
+    /**
+     * Unbans a user from a Group.
+     * @param param0 The parameters for the request.
+     * @returns The GroupMember object of the unbanned user.
+     */
+    //! Same here
+    public async unbanGroupMember({
+        groupId,
+        userId
+    }: VRCAPI.Groups.Requests.UnbanGroupMemberRequest): Promise<VRCAPI.Groups.Models.GroupMember> {
+
+        if (userId.length <= 0) return Promise.reject(new Error('userId is too short!'));
+
+        const paramRequest: VRCAPI.Generics.executeRequestType = {
+            currentRequest: ApiPaths.groups.unbanGroupMember,
+            pathFormated: ApiPaths.groups.unbanGroupMember.path.replace('{groupId}', groupId).replace('{userId}', userId),
+        };
+
+        this.checkValidData(paramRequest);
+        const queryResult = await this.executeRequest<VRCAPI.Groups.Models.GroupMember>(paramRequest);
+
+        return queryResult;
+
+    }
+
+    /**
+     * Creates a gallery for a Group.
+     * @param param0 The parameters for the request.
+     * @returns A GroupGallery object.
+     */
+    public async createGroupGallery({
+        groupId,
+        name, // min 1 character
+        description,
+        membersOnly,
+        roleIdsToView,
+        roleIdsToSubmit,
+        roleIdsToAutoApprove,
+        roleIdsToManage
+    }: VRCAPI.Groups.Requests.createGroupGalleryRequest): Promise<VRCAPI.Groups.Models.GroupGallery> {
+
+        if (name && name.length < 1) return Promise.reject(new Error('Name must be at least 1 character!'));
+
+        const body: VRCAPI.Generics.dataKeysGroupCreateGallery = {
+            name: name,
+        }
+        
+        // check parameters
+        if (description && description.length >= 0) {
+            body.description = description;
+        }
+
+        if (membersOnly) {
+            body.membersOnly = membersOnly;
+        }
+
+        if (roleIdsToView) {
+            body.roleIdsToView = roleIdsToView;
+        }
+
+        if (roleIdsToSubmit) {
+            body.roleIdsToSubmit = roleIdsToSubmit;
+        }
+
+        if (roleIdsToAutoApprove) {
+            body.roleIdsToAutoApprove = roleIdsToAutoApprove;
+        }
+
+        if (roleIdsToManage) {
+            body.roleIdsToManage = roleIdsToManage;
+        }
```

Comment:
You can compose this differently, remember that anything that is passed and is undefined is just going to be empty when create an object with properties.